src/server/server.cpp

```diff
@@ -1779,8 +1779,11 @@ void server::handle_player_in_game(socket_ptr socket, std::shared_ptr<simple_wml
 			const std::vector<std::string> mods = utils::split(m["active_mods"].to_string());
 			const std::vector<std::string> mod_sources = utils::split(m["active_mod_versions"].to_string());
 			const std::vector<std::string> mod_versions = utils::split(m["active_mod_addon_ids"].to_string());
-			if(mods.size() != 0) {
-				for(unsigned int i = 0; i < mods.size(); i++){
+			const bool mod_info_present = mods.size() == mod_sources.size() && mod_sources.size() == mod_versions.size();
+			for(unsigned int i = 0; i < mods.size(); i++){
+				if(!mod_info_present) {
+					user_handler_->db_insert_modification_info(uuid_, g.db_id(), mods[i], "", "");
```

Comment:
Are these inserts useful? Perhaps just log a warning about inconsistent mod info?

Reply:
Ah wait, `active_mods` contains something, ok. I suppose we can investigate inconsistent mod info from the DB then. No need for a log message.

Reply:
Might still be useful to log what is in `active_mod_versions` and `active_mod_addon_ids`.

Reply:
Though would it be useful, given that at the moment every released 1.14 client would be logging it?

Reply:
Because versions and ids are empty? We wouldn’t need to log those since there’s no additional info there.

Reply:
What would be the need to add logging here, then?

Reply:
I'm saying that it might be useful to log what is in `active_mod_versions` and `active_mod_addon_ids` when the sizes are not all equal, so we can figure out if something goes wrong and hopefully also see why.

My assumption is that when you said "at the moment every released 1.14 client would be logging" it means those two keys are empty. In that case logging would not be useful though so that case should be excluded from logging.

Reply:
Alright, updated.

Reply:
I think the logging should be outside the loop? Also please add the game IDs so we can reference the DB entry to the log message.

Something like "mod info mismatch in game ... mods: ..."

Reply:
Updated.